tesseract/tesseract_planning/tesseract_motion_planners/examples/freespace_example.cpp

```diff
@@ -0,0 +1,235 @@
+    plotter->init(tesseract);
+
+  ManipulatorInfo manip;
+  manip.manipulator = "manipulator";
+  manip.manipulator_ik_solver = "OPWInvKin";
+
+  opw_kinematics::Parameters<double> opw_params;
+  opw_params.a1 = (0.100);
+  opw_params.a2 = (-0.135);
+  opw_params.b = (0.000);
+  opw_params.c1 = (0.615);
+  opw_params.c2 = (0.705);
+  opw_params.c3 = (0.755);
+  opw_params.c4 = (0.085);
+
+  opw_params.offsets[2] = -M_PI / 2.0;
+
+  auto robot_kin = tesseract->getFwdKinematicsManagerConst()->getFwdKinematicSolver(manip.manipulator);
+  auto opw_kin = std::make_shared<tesseract_kinematics::OPWInvKin>();
+  opw_kin->init(manip.manipulator,
+                opw_params,
+                robot_kin->getBaseLinkName(),
+                robot_kin->getTipLinkName(),
+                robot_kin->getJointNames(),
+                robot_kin->getLinkNames(),
+                robot_kin->getActiveLinkNames(),
+                robot_kin->getLimits());
+
+  tesseract->getInvKinematicsManager()->addInvKinematicSolver(opw_kin);
+  tesseract->getInvKinematicsManager()->setDefaultInvKinematicSolver(manip.manipulator, opw_kin->getSolverName());
+
+  auto fwd_kin = tesseract->getFwdKinematicsManagerConst()->getFwdKinematicSolver(manip.manipulator);
+  auto inv_kin = tesseract->getInvKinematicsManagerConst()->getInvKinematicSolver(manip.manipulator);
+  auto cur_state = tesseract->getEnvironmentConst()->getCurrentState();
+
+  // Specify start location
+  JointWaypoint wp0 = Eigen::VectorXd::Zero(6);
+
+  // Specify freespace start waypoint
+  CartesianWaypoint wp1 =
+      Eigen::Isometry3d::Identity() * Eigen::Translation3d(0.8, -.20, 0.8) * Eigen::Quaterniond(0, 0, -1.0, 0);
+
+  // Define Plan Instructions
+  PlanInstruction start_instruction(wp0, PlanInstructionType::START);
+  PlanInstruction plan_f1(wp1, PlanInstructionType::FREESPACE, "DEFAULT");
+
+  // Create program
+  CompositeInstruction program;
+  program.setStartInstruction(start_instruction);
+  program.setManipulatorInfo(manip);
+  program.push_back(plan_f1);
+
+  // Plot Program
+  if (plotter)
+  {
+  }
+
+  // Create Profiles
+  auto ompl_plan_profile = std::make_shared<OMPLDefaultPlanProfile>();
+  auto trajopt_plan_profile = std::make_shared<TrajOptDefaultPlanProfile>();
+  auto trajopt_composite_profile = std::make_shared<TrajOptDefaultCompositeProfile>();
+
+  // Create a seed
+  CompositeInstruction seed = generateSeed(program, cur_state, tesseract);
+
+  // Create Planning Request
+  PlannerRequest request;
+  request.seed = seed;
+  request.instructions = program;
+  request.tesseract = tesseract;
+  request.env_state = cur_state;
+
+  // Solve OMPL Plan
+  PlannerResponse ompl_response;
+  OMPLMotionPlanner ompl_planner;
+  ompl_planner.plan_profiles["DEFAULT"] = ompl_plan_profile;
+  ompl_planner.problem_generator = tesseract_planning::DefaultOMPLProblemGenerator;
+  auto ompl_status = ompl_planner.solve(request, ompl_response);
+  assert(ompl_status);
+
+  // Plot Descartes Trajectory
```

Comment:
I believe this is supposed to be OMPL trajectory

Reply:
Fixed.